### PR TITLE
ocamlPackages.ocaml_gettext: 0.3.8 → 0.4.1

### LINKED ISSUE #84416: Package vagrant broken: camlp4 is not available for OCaml 4.09.1
**Describe the bug**
```
in job ‘vagrant.x86_64-linux’:
camlp4 is not available for OCaml 4.09.1
```

**To Reproduce**
1. Build vagrant on latest master
2. Get error

**Expected behavior**
Builds

**Screenshots**

**Additional context**

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.0-72-generic, Ubuntu, 18.04.4 LTS (Bionic Beaver)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(maciej): `"nixos-20.09pre218613.ae6bdcc5358, nixpkgs-20.09pre218717.05f0934825c"`
 - nixpkgs: `/home/maciej/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - vagrant
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

Do not depend on deprecated camlp4

Fixes #84416

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vbgl** [comment]
@GrahamcOfBorg build libguestfs libguestfs-with-appliance python27Packages.guestfs python37Packages.guestfs python38Packages.guestfs vagrant virt-top

File: pkgs/development/ocaml-modules/ocaml-gettext/stub.nix [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{ lib, buildDunePackage, ocaml_gettext, ounit }:
+
+buildDunePackage rec {
+
+  pname = "gettext-stub";
```
Comment (timokau) [inline pkgs/development/ocaml-modules/ocaml-gettext/stub.nix:5]: What is the point of this package?

File: pkgs/development/ocaml-modules/ocaml-gettext/default.nix [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
 
-  createFindlibDestdir = true;
+  checkInputs = lib.optional doCheck ounit;
 
   dontStrip = true;
 
-  meta = with stdenv.lib; {
+  meta = with lib; {
     description = "OCaml Bindings to gettext";
-    homepage = https://forge.ocamlcore.org/projects/ocaml-gettext;
```
Comment (timokau) [inline pkgs/development/ocaml-modules/ocaml-gettext/default.nix:33]: Was there an ownership change?

File: pkgs/development/ocaml-modules/ocaml-gettext/default.nix [hunk 3]
```diff
@@ -1,38 +1,28 @@
...
-    license = licenses.gpl2;
+    homepage = "https://github.com/gildor478/ocaml-gettext";
+    license = licenses.lgpl21;
     maintainers = [ maintainers.volth ];
-    platforms = ocaml.meta.platforms or [];
```
Comment (timokau) [inline pkgs/development/ocaml-modules/ocaml-gettext/default.nix:36]: Why was this removed?

File: pkgs/development/ocaml-modules/ocaml-gettext/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/ocaml-gettext/default.nix:33]: If you go there: http://forge.ocamlcore.org/projects/ocaml-gettext (no HTTPS), you can read: “Project has moved to Github: https://github.com/gildor478/ocaml-gettext”

File: pkgs/development/ocaml-modules/ocaml-gettext/default.nix (hunk 3 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/ocaml-gettext/default.nix:36]: Because it is set by `buildDunePackage`.

File: pkgs/development/ocaml-modules/ocaml-gettext/stub.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/ocaml-gettext/stub.nix:5]: It’s a library that is needed to build `libguestfs` and `virt-top`.

Comment (timokau) [reply pkgs/development/ocaml-modules/ocaml-gettext/stub.nix:5]: How does it differ from the normal `gettext`? It looks identical to me.

Comment (vbgl) [reply pkgs/development/ocaml-modules/ocaml-gettext/stub.nix:5]: The `pname` is different. You can have a look to the source code of `buildDunePackage`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/ocaml/dune.nix#L16 and to its documentation: https://nixos.org/nixpkgs/manual/#sec-language-ocaml

Comment (timokau) [reply pkgs/development/ocaml-modules/ocaml-gettext/stub.nix:5]: Oh, I didn't expect the `pname` to have an impact on the build result. Sorry for the noise. I'm not familiar with ocaml, only reviewing this to speed up the progress :)

**Review by timokau — APPROVED** [review]
Looks good to me.
